packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolGetEntity.ts

```diff
@@ -57,7 +50,21 @@ export const useBlockProtocolGetEntity = (): {
 
       const [{ data: entityResponseData }, { data: outgoingLinksData }] =
         await Promise.all([
-          getEntityFn({ variables: { entityId } }),
+          getEntityFn({
+            variables: {
+              entityId,
+              // Get the full entity type _tree_
+              dataTypeResolveDepth: 255,
+              propertyTypeResolveDepth: 255,
+              linkTypeResolveDepth: 255,
+              // Don't explore entityType references beyond the absolute neighbors
+              /** @todo - do we want to use 1 or 2 here? */
+              entityTypeResolveDepth: 2,
+              // Only get absolute neighbor entities
+              linkResolveDepth: 2,
+              linkTargetEntityResolveDepth: 2,
```

Comment:
Yes, I was thinking that last paragraph might be the case. @TimDiekmann what would happen here?

Reply:
`linkResolveDepth` will resolve links from an entity, `linkTargetEntityResolveDepth` will resolve the target entity of a link. Given the following chain:
```
Entity A -> Link A -> Entity B -> Link B -> Entity C
```
- a `linkResolveDepth` of `1` will resolve `Entity A -> Link A`
- a `linkTargetEntityResolveDepth ` of `1` will resolve `Link A -> Entity B`
- a `linkResolveDepth` of `2` will resolve `Entity B -> Link B`
- a `linkTargetEntityResolveDepth ` of `2` will resolve `Link B -> Entity C`

So if `linkTargetEntityResolveDepth` is `0`, nothing will be resolved as `Link A` is not resolved at all and if `linkResolveDepth ` is `2` and `linkTargetEntityResolveDepth` is `0` only `Link A` will be resolved.

Reply:
Thank you, Tim. This does answer this concretely. @Alfred-Mountfield we may want to change the values to reflect this, then. I suppose we just want `1` instead of `2` here

Reply:
Just to flag this has been reflected in #1263